web_ui/frontend/app/registry/namespace/components/NamespaceForm.tsx

```diff
@@ -0,0 +1,174 @@
+            const response = await fetch(url)
+            if (response.ok) {
+                const responseData: Institution[] = await response.json()
+                setInstitutions(responseData)
+            }
+        })()
+    }, []);
+
+    const onSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+
+        const form = e.currentTarget
+
+        const successfulSubmit = await handleSubmit(e)
+
+        // Clear the form on successful submit
+        if (successfulSubmit) {
+            form.reset()
+            setInstitution("")
+            setServerType("")
+        }
+    }
+
+    return (
+        <form onSubmit={onSubmit}>
+            <Box pb={2}>
+                <TextField
+                    fullWidth
+                    size={"small"}
+                    id={"prefix"}
+                    name={"prefix"}
+                    label={"Prefix"}
+                    defaultValue={namespace?.prefix || ""}
+                    variant={"outlined"}
+                    onChange={event => {
+                        if (event.target.value == "") {
+                            setServerType("")
+                        } else if (event.target.value.startsWith("/cache")) {
+                            setServerType("cache")
+                        } else {
+                            setServerType("origin")
+                        }
+                    }}
+                />
+            </Box>
+            <Box pb={2}>
+                <FormControl fullWidth size={"small"}>
+                    <InputLabel id="institution-label">Namespace Type</InputLabel>
+                    <Select
+                        labelId="institution-label"
+                        id="institution"
+                        label="Institution"
+                        value={serverType}
+                        inputProps={{readOnly: true}}
+                    >
+                        <MenuItem value={"cache"}>Cache</MenuItem>
+                        <MenuItem value={"origin"}>Origin</MenuItem>
+                    </Select>
+                    <FormHelperText>Read Only: Caches are declared with a &apos;/cache&apos; prefix</FormHelperText>
+                </FormControl>
+            </Box>
+            <Box pb={2}>
+                <TextField
+                    fullWidth
+                    size={"small"}
+                    id={"pubkey"}
+                    name={"pubkey"}
+                    label={"Pubkey"}
+                    variant={"outlined"}
+                    multiline={true}
+                    defaultValue={namespace?.pubkey || ""}
+                    inputProps={{
+                        style: {
+                            fontFamily: "monospace",
+                            fontSize: "0.8rem",
+                            lineHeight: "0.9rem",
+                            minHeight: "1.5rem"
+                        }
+                    }}
```

Comment:
Can we also add a helper text here? "Pubkey is your origin's public JWKS. It must in JSON format, not JSON string".

If possible, I'd like to have a placeholder value to make it clear:
```json
{
    "keys": [
        {
            "alg": "ES256",
            "crv": "P-256",
            "kid": "<Your-kid>",
            "kty": "EC",
            "x": "",
            "y": ""
        }
    ]
}
```